server/service/osquery.go

```diff
@@ -986,7 +988,8 @@ func (svc *Service) SubmitDistributedQueryResults(
 		return ctxerr.Wrap(ctx, err, "getting app config")
 	}
 
-	if len(labelResults) > 0 {
+	// If all queries failed, we do not want to update labels because the "All Hosts" label must always pass, and we should not be deleting it.
+	if havePassingQuery && len(labelResults) > 0 {
```

Comment:
I'm not following the issue fully and am not seeing how this fixes such issue.

`RecordLabelQueryExecutions` only deletes labels that the host failed to execute or that executed successfully and returned no results.

Do we know why the "All hosts" or "All Chrome hosts" labels are failing or returning no results?

Reply:
Yes, fleetd-chrome has a bug where all queries start failing. This doesn't fix that issue but just ensures labels aren't cleared if all query results come back with error status. I'm only fixing labels here because that was the obvious issue I saw.

Reply:
I will continue fixing the issue on fleetd-chrome -- that will be a separate PR.

Reply:
I think we should fix this on the fleetd-chrome side. We do want to continuing to clear labels if they fail to execute their query, and not have edge cases for fleetd-chrome (if possible).